libsolidity/ast/Types.cpp

```diff
@@ -476,7 +476,7 @@ bool IntegerType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 	return _convertTo.category() == category() ||
 		_convertTo.category() == Category::Contract ||
 		_convertTo.category() == Category::Enum ||
-		_convertTo.category() == Category::FixedBytes ||
+		(_convertTo.category() == Category::FixedBytes && storageBytes() == _convertTo.storageBytes()) ||
```

Comment:
`storageBytes` is a deduced property and might change in the future. IMHO it is better to use the native functions and mulitply by 8.